src/util/proteomecluster.cpp

```diff
@@ -0,0 +1,492 @@
+        : proteomeKey(proteomeKey), proteomeAALen(pLength), repProtKey(repKey), protSimScore(simScore), memCount(memCount), totalSeqId(seqId) {}
+
+    void incrementMemCount(unsigned int count) {
+        memCount += count;
+    }
+    void decrementmemCount() {
+        memCount--;
+    }
+    int getRepProtKey() {
+        return repProtKey;
+    }
+    int getProteomeKey() {
+        return proteomeKey;
+    }
+    bool isCovered(){
+        if (repProtKey == -1) {
+            return false;
+        }
+        return true;
+    }
+    void computeRedundancy() {
+        protSimScore = totalSeqId / proteomeAALen;
+
+    }
+    void addSeqId(float seqId) {
+        totalSeqId += seqId;
+    }
+    void addSeqLen(unsigned int seqLength) {
+        proteomeAALen += seqLength;
+    }
+    void resetProteomeInfo(){
+        memCount = 0;
+        totalSeqId = 0.0f;
+    }
+    float getProtSimScore() {
+        return protSimScore;
+    }
+    static bool compareByKey(const ProteomeEntry& a, const ProteomeEntry& b) {
+        
+        if (a.repProtKey < b.repProtKey){
+            return true;
+        }
+        if (a.repProtKey > b.repProtKey){
+            return false;
+        }
+        if (a.proteomeKey == a.repProtKey && b.proteomeKey != b.repProtKey){
+            return true;
+        }
+        if (a.proteomeKey != a.repProtKey && b.proteomeKey == b.repProtKey){
+            return false;
+        }
+        return false;
+    }
+};
+
+struct __attribute__((__packed__)) MemberProteinEntry{
+    unsigned int proteomeKey;
+    unsigned int proteinKey;
+};
+
+struct ClusterEntry {
+    bool isAvailable;
+    std::vector<MemberProteinEntry> memberProteins;
+
+    ClusterEntry() : isAvailable(false) {}
+
+    ClusterEntry(size_t memberProteinSize) : isAvailable(true) {
+        memberProteins.reserve(memberProteinSize);
+    }
+
+    void resetClusterInfo(){
+        memberProteins.clear();
+        isAvailable = false;
+    }
+};
+
+void calculateProteomeLength(std::vector<ProteomeEntry>& ProteomeList, DBReader<unsigned int>::LookupEntry* const& lookup, size_t lookupSize, DBReader<unsigned int>& tProteinDB) {
+    for (size_t i = 0; i < lookupSize; i++) {
+        const unsigned int ProteomeId = lookup[i].fileNumber;
+        const unsigned int ProteinId = lookup[i].id;
+        ProteomeList[ProteomeId].addSeqLen(tProteinDB.getSeqLen(ProteinId));
+        if (ProteomeList[ProteomeId].proteomeKey == -1){
+            ProteomeList[ProteomeId].proteomeKey = ProteomeId;
+        }
+    }
+}
+
+void initLocalClusterReps(size_t& id, std::vector<ClusterEntry>& localClusterReps, std::vector<unsigned int>& localMemCount, DBReader<unsigned int>& tProteinDB, DBReader<unsigned int>& linResDB, unsigned int thread_idx){
+    std::vector<unsigned int> memberKeys;
+    memberKeys.reserve(50); // store key for every protein in a cluster
+
+    std::vector<bool> isProteomeInCluster(localMemCount.size(), false); ;
```

Comment:
double `;`

also careful about bool vectors. They are not multi-threading friendly. Here its fine, as you don't do any multi-threading